packages/ec-core/src/distributions/wrappers/choose_cloning.rs

```diff
@@ -1,33 +1,77 @@
 use std::num::NonZeroUsize;
 
-use miette::Diagnostic;
-use rand::{distr::slice::Choose, prelude::Distribution};
+use rand::{
+    distr::slice::{Choose, Empty},
+    prelude::Distribution,
+};
 
-use crate::distributions::choices::ChoicesDistribution;
+use crate::distributions::finite::Finite;
 
-/// Generate a random element from an array of options, cloning the choosen
-/// element.
-#[derive(Debug, Clone, Copy)]
+/// Uniform [`Distribution`] of a slice of options, cloning the chosen element,
+/// borrowing from a collection of choices.
+///
+/// This [`Distribution`] borrows the collection.
+///
+/// Also see [`ChooseCloningOwning`](super::choose_cloning_owning::ChooseCloningOwning) for an
+/// alternative that takes ownership of the collection instead.
+///
+///
+/// # Example
+/// ```
+/// # use rand::{rng, distr::{Distribution, slice::Empty}};
+/// # use ec_core::distributions::wrappers::choose_cloning::ChooseCloning;
+/// #
+/// # fn main() -> Result<(), Empty> {
+/// let distribution = ChooseCloning::new(&[1, 2, 3])?;
```

Comment:
I wouldn't want to block the PR for this, but would it make sense to have the example use types where cloning might matter (like `String`) instead of using a type that also implements `Copy`?

This also applies to some of the other examples below.